.github/workflows/docker-publish.yml

```diff
@@ -54,22 +51,32 @@ jobs:
         run: |
           KIND_VERSION=$(cat KIND_VERSION)
           BASE_IMAGE=$(cat BASE_IMAGE)
+          KUBERNETES_VERSION=$(curl -Ls https://dl.k8s.io/release/stable.txt)
           echo "KIND_VERSION=$KIND_VERSION" >> $GITHUB_ENV
           echo "BASE_IMAGE=$BASE_IMAGE" >> $GITHUB_ENV
+          echo "KUBERNETES_VERSION=$KUBERNETES_VERSION" >> $GITHUB_ENV
 
-      - name: Install kind
+      - name: Build and install kind - ${{ env.KIND_VERSION }}
         run: |
-          curl -Lo ./kind https://kind.sigs.k8s.io/dl/${{ env.KIND_VERSION }}/kind-linux-amd64
-          chmod +x ./kind
-          sudo mv ./kind /usr/local/bin/kind
+          # The runner has kind pre-installed, removing the same to download the preferred version.
+          rm -rf $(which kind)
```

Comment:
why it will have the previous kind?

Reply:
During builds, it was observed that 
```
      - name: Build and install kind - ${{ env.KIND_VERSION }}
        run: |
          # The runner has kind pre-installed, removing the same to download the preferred version.
          ls -ltra $(which kind)
          kind --version
  ```
  Logs from workflow
  ```
2024-06-26T12:57:34.3052575Z -rwxrwxrwx 1 root root 6534249 Jun 16 21:36 /usr/local/bin/kind
2024-06-26T12:57:34.3278786Z kind version 0.23.0
```

checked a little further:
ls -ltra /usr/local/bin
```
-rwxrwxrwx  1 root   root    88652584 Jun 16 21:33 stack
-rwxrwxrwx  1 root   root     6534249 Jun 16 21:36 kind    <-----
-rwxrwxrwx  1 root   root     445336 Jun 16 21:36 helm
-rwxrwxrwx  1 root   root    95595640 Jun 16 21:36 minikube
-rwxrwxrwx  1 root   root    15048704 Jun 16 21:36 kustomize
-rwxrwxrwx  1 root   root       14217 Jun 16 21:36 lein
```
Maybe they all are pre-installed. 

Despite building the kind binary with the changes that point to the image we may need to use, by precedence, the default pre-installed binary is picked, which builds the node images leading to certain errors when the built node-image is used.

Reply:
The current observation is:

Pre-installation (which is picked due to precedence in PATH)
```
-rwxrwxrwx 1 root root 6534249 Jun 16 21:36 /usr/local/bin/kind
```

Post removal and reinstallation with path set to $GOPATH/bin
```
-rwxr-xr-x 1 runner docker 6534249 Jun 26 13:48 /home/runner/go/bin/kind
```